### ISSUE TITLE
Aldjemy release contains unnecessary components like tests

### ISSUE THREAD

**Issue by Mogost** [aldjemy/aldjemy#162]
![image](https://user-images.githubusercontent.com/721963/118987052-94519f00-b988-11eb-99d2-3d3603e4f8b6.png)

**Comment by ryanhiebert** [comment]
So, I actually don't mind having the tests ship along with the package. I don't want them to run, but it's not important to remove them to me. Strictly speaking, I wouldn't be opposed to removing them, but so far in Python I haven't been able to figure out how to accomplish some other goals while keeping the tests out of the directory. Particularly having tests co-located right alongside the relevant modules in a package, which I intend to work toward.

So in summary, I'm not opposed to removing some files from the package in principle, but beware if you attempt it that I'm going to be picky to make sure it doesn't cause any problems with other of my priorities and preferences.

**Comment by ryanhiebert** [comment]
Feel free to re-open this if you think it's worth your time to try to do this. I don't expect to be doing this myself.

**Comment by luzfcb** [comment]
@ryanhiebert if you can create a new branch 1.0.x from 1.0.0 tag and:

1 - Add on setup.py:

```python
    packages=find_packages(exclude=['test_project', 'test_project_postgres', 'test_project.*', 'test_project_postgres.*']),
    include_package_data=True,
```


> Note: `include_package_data=True` will force read the rules from MANIFEST.in file when run `sdist`.
   the `exclude` in `find_packages` will force not to include the excluded packages in the `wheel` package.

#### setup.py

```python
#!/usr/bin/env python

import os.path
from setuptools import setup, find_packages


def read(fname):
    return open(os.path.join(os.path.dirname(__file__), fname)).read()


setupconf = dict(
    name="aldjemy",
    version="1.0.1",
    license="BSD",
    url="https://github.com/aldjemy/aldjemy/",
    author="Mihail Krivushin",
    author_email="krivushinme@gmail.com",
    description="SQLAlchemy to Django integration library",
    long_description=read("README.rst"),
    packages=find_packages(exclude=['test_project', 'test_project_postgres', 'test_project.*', 'test_project_postgres.*']),
    include_package_data=True,
    install_requires=["sqlalchemy>0.7.1"],
    classifiers=[
        "Intended Audience :: Developers",
        "License :: OSI Approved :: BSD License",
        "Operating System :: OS Independent",
        "Programming Language :: Python",
        "Programming Language :: Python :: 3",
        "Programming Language :: Python :: 3.7",
        "Programming Language :: Python :: 3.8",
        "Framework :: Django",
        "Framework :: Django :: 2.2",
        "Framework :: Django :: 3.0",
        "Framework :: Django :: 3.1",
    ],
)

if __name__ == "__main__":
    setup(**setupconf)

```


2 - Change the MANIFEST.in to

```python
prune test*
include MANIFEST.in
include README.rst
include LICENSE
graft aldjemy
global-exclude *.py[cod]
```


Build the packages:

```
python3 setup.py clean --all && python3 setup.py sdist
python3 setup.py clean --all && python3 setup.py bdist_wheel
```

and release the 1.0.1 version. (only if you want)

**Comment by ryanhiebert** [comment]
Thanks, @luzfcb , that's quite a write-up. Thank you for being willing to spell that out!

At this point I'm really wanting to stick with Poetry, rather than using setuptools. For unit tests, I'd like to be able to do relative imports, and have co-located names. We don't do that currently, but I find that to be a good pattern. I don't think that would be reasonable if we exclude the tests from the distribution. I have deleted some of the other files that aren't needed, and I'm continuing some janitorial work to simplify things in this project.

Thanks again!

**Comment by luzfcb** [comment]
@ryanhiebert the main branch (that has poetry config) doesn't include the test* packages on the `sdist` and `wheel` packages. (I tested it manually), 
But version 1.0.0 still doesn't include poetry config and it why I proposed the changes on `setup.py` and `MANIFEST.in`. I just wanted to make your life easier with as little change as possible. 

that is, this change only applies to a possible new branch starting from 1.0.0, which would be used solely and exclusively to remove the test packages and make the manual release of version 1.0.1 without those packages.

**Comment by ryanhiebert** [comment]
I see. Is a 1.0.1 release important to you for some reason? Or is the 2.0 package good for what you're trying to do? I'd prefer to not create an additional 1.0.x release if we don't need to.

**Comment by luzfcb** [comment]
I don't really need 1.0.1, however, I don't consider it a good practice to send unnecessary code as a test package.
The problem here is that the `test_*` module directory will be installed at the same directory level as the `aldjemy` module directory.
In the past, I've had hell with strange errors in my tests and it was annoying to discover that another django package sent a test package with the same name as a library with testing tools that I used. 

anyway, I don't use 1.0.0. I just wanted to help quickly resolve this issue (which currently exists in 1.0.0)

**Comment by ryanhiebert** [comment]
Cool. As long as the top-level directory issue is solved with 2.0, I'm happy.
